Marlin/src/HAL/HAL_AVR/endstop_interrupts.h

```diff
@@ -107,7 +107,7 @@ void setup_endstop_interrupts( void ) {
       attachInterrupt(digitalPinToInterrupt(X_MAX_PIN), endstop_ISR, CHANGE); // assign it
     #else
       // Not all used endstop/probe -pins can raise interrupts. Please deactivate ENDSTOP_INTERRUPTS or change the pin configuration!
-      static_assert(digitalPinToPCICR(X_MAX_PIN) != NULL, "X_MAX_PIN is not interrupt-capable"); // if pin has no pin change interrupt - error
+      static_assert(digitalPinToPCICR(X_MAX_PIN) != nullptr, "X_MAX_PIN is not interrupt-capable"); // if pin has no pin change interrupt - error
```

Comment:
Those are pointers:
```
  55   #define digitalPinToPCICR(p)    ( WITHIN(p, 10, 15) || \
  56                                     WITHIN(p, 50, 53) || \
  57                                     WITHIN(p, 62, 69) ? &PCICR : (uint8_t*)0 )
```
The macro either returns ``&PCICR`` (pointer) or ``(uint8_t*)0`` (pointer). It's arguable that we should also replace the ``(uint8_t*)0`` (and any other occurrences of ``(<type>*)0``) with ``nullptr`` as well.

Reply:
hmm right, sorry, the git grep doesnt work on multiple lines :p

buildroot/share/pin_interrupt_test/pin_interrupt_test.ino:6:    #define digitalPinToPCICR(p)    ( ((p) >= 10 && (p) <= 15) || \

Reply:
I think it is incorrect to define pin numbers as 'nullptr' if they don't exist.   Pin numbers are not pointers.    In this case...   'NULL' seems more accurate and correct:
```
- #define DIO0_PWM    NULL
+ #define DIO0_PWM    nullptr

- #define DIO1_PWM    NULL
+ #define DIO1_PWM    nullptr
```

Reply:
I think it would be a better option in the "nonexistent pin" case to define our own constant (or just use 0) rather than use NULL. C++ 11 allows compilers to define NULL as nullptr instead of as 0, so it looks like the C++ standard is trying to get away from allowing the use of NULL in non-pointer contexts.

Reply:
> so it looks like the C++ standard is trying to get away from allowing the use of NULL in non-pointer contexts.

You can't just outlaw the use of word like Null.    There are Null Lists.  There are Null Sets.  There are Null Vectors.    You can't just say "We are changing what a word means."

Reply:
In some fastio headers the `_PWM` fields specify the address of PWM timers. In others the fields are direct timer references (such as `OCR1A`). I have updated the fastio headers here so that the `_PWM` zero values are compatible with the `_PWM` non-zero values.